#
# OpenSSH 8.2 delivered support for FIDO/U2F hardware authenticators.
# OpenSSH client side of FIDO support depends on libfido2 library
# and its dependencies (currently libcbor library, but there might
# be more in future) which is currently not available in Solaris.
# Thus, the patch removes mentions of FIDO/U2F suppport
# from OpenSSH client man pages. The server side remains FIDO capable
# which means that the server accepts FIDO keys.
#
# As such, this is Solaris specific patch.
#
--- old/ssh-keygen.1
+++ new/ssh-keygen.1
@@ -51,7 +51,7 @@
 .Op Fl m Ar format
 .Op Fl N Ar new_passphrase
 .Op Fl O Ar option
-.Op Fl t Cm ecdsa | ecdsa-sk | ed25519 | ed25519-sk | mldsa44-ed25519 | rsa
+.Op Fl t Cm ecdsa | ed25519 | mldsa44-ed25519 | rsa
 .Op Fl w Ar provider
 .Op Fl Z Ar cipher
 .Nm ssh-keygen
@@ -207,9 +207,7 @@ Normally each user wishing to use SSH
 with public key authentication runs this once to create the authentication
 key in
 .Pa ~/.ssh/id_ecdsa ,
-.Pa ~/.ssh/id_ecdsa_sk ,
 .Pa ~/.ssh/id_ed25519 ,
-.Pa ~/.ssh/id_ed25519_sk ,
 .Pa ~/.ssh/id_mldsa44-ed25519
 or
 .Pa ~/.ssh/id_rsa .
@@ -305,7 +303,7 @@ flag determines the key length by selecting from one of three elliptic
 curve sizes: 256, 384 or 521 bits.
 Attempting to use bit lengths other than these three values for ECDSA keys
 will fail.
-ECDSA-SK, Ed25519 and Ed25519-SK keys have a fixed length and the
+Ed25519 key has a fixed length and the
 .Fl b
 flag will be ignored.
 .It Fl C Ar comment
@@ -582,13 +580,11 @@ by key ID or serial number.
 See the
 .Sx KEY REVOCATION LISTS
 section for details.
-.It Fl t Cm ecdsa | ecdsa-sk | ed25519 | ed25519-sk | mldsa44-ed25519 | rsa
+.It Fl t Cm ecdsa | ed25519 | mldsa44-ed25519 | rsa
 Specifies the type of key to create.
 The possible values are
 .Dq ecdsa ,
-.Dq ecdsa-sk ,
 .Dq ed25519 (the default),
-.Dq ed25519-sk ,
 .Dq mldsa44-ed25519
 or
 .Dq rsa .
@@ -698,7 +694,7 @@ The maximum is 3.
 .It Fl w Ar provider
 Specifies a path to a library that will be used when creating
 FIDO authenticator-hosted keys, overriding the default of using
-the internal USB HID support.
+the internal USB HID support.  Not supported in Solaris.
 .It Fl Y Cm find-principals
 Find the principal(s) associated with the public key of a signature,
 provided using the
@@ -1048,7 +1044,8 @@ of user presence (e.g. by having the user touch the authenticator).
 This option only makes sense for the FIDO authenticator algorithms
 .Cm ecdsa-sk
 and
-.Cm ed25519-sk .
+.Cm ed25519-sk
+which are currently not supported on Solaris.
 .Pp
 .It Ic source-address Ns = Ns Ar address_list
 Restrict the source addresses from which the certificate is considered valid.
@@ -1298,14 +1295,12 @@ user2@example.com namespaces="file" ssh-ed25519 AAA41...
 .It Ev SSH_SK_PROVIDER
 Specifies a path to a library that will be used when loading any
 FIDO authenticator-hosted keys, overriding the default of using
-the built-in USB HID support.
+the built-in USB HID support.  Not supported in Solaris.
 .El
 .Sh FILES
 .Bl -tag -width Ds -compact
 .It Pa ~/.ssh/id_ecdsa
-.It Pa ~/.ssh/id_ecdsa_sk
 .It Pa ~/.ssh/id_ed25519
-.It Pa ~/.ssh/id_ed25519_sk
 .It Pa ~/.ssh/id_mldsa44-ed25519
 .It Pa ~/.ssh/id_rsa
 Contains the ECDSA, authenticator-hosted ECDSA, Ed25519,
@@ -1321,9 +1316,7 @@ but it is offered as the default file for the private key.
 will read this file when a login attempt is made.
 .Pp
 .It Pa ~/.ssh/id_ecdsa.pub
-.It Pa ~/.ssh/id_ecdsa_sk.pub
 .It Pa ~/.ssh/id_ed25519.pub
-.It Pa ~/.ssh/id_ed25519_sk.pub
 .It Pa ~/.ssh/id_mldsa44-ed25519.pub
 .It Pa ~/.ssh/id_rsa.pub
 Contains the ECDSA, authenticator-hosted ECDSA, Ed25519,
--- old/ssh.1
+++ new/ssh.1
@@ -306,10 +306,8 @@ when the private key file is not present locally.
 The default is
 .Pa ~/.ssh/id_rsa ,
 .Pa ~/.ssh/id_ecdsa ,
-.Pa ~/.ssh/id_ecdsa_sk ,
-.Pa ~/.ssh/id_ed25519
 and
-.Pa ~/.ssh/id_ed25519_sk .
+.Pa ~/.ssh/id_ed25519
 Identity files may also be specified on
 a per-host basis in the configuration file.
 It is possible to have multiple
@@ -985,24 +983,16 @@ The user creates their key pair by running
 This stores the private key in
 .Pa ~/.ssh/id_ecdsa
 (ECDSA),
-.Pa ~/.ssh/id_ecdsa_sk
-(authenticator-hosted ECDSA),
 .Pa ~/.ssh/id_ed25519
 (Ed25519),
-.Pa ~/.ssh/id_ed25519_sk
-(authenticator-hosted Ed25519),
 or
 .Pa ~/.ssh/id_rsa
 (RSA)
 and stores the public key in
 .Pa ~/.ssh/id_ecdsa.pub
 (ECDSA),
-.Pa ~/.ssh/id_ecdsa_sk.pub
-(authenticator-hosted ECDSA),
 .Pa ~/.ssh/id_ed25519.pub
 (Ed25519),
-.Pa ~/.ssh/id_ed25519_sk.pub
-(authenticator-hosted Ed25519),
 or
 .Pa ~/.ssh/id_rsa.pub
 (RSA)
@@ -1599,9 +1589,7 @@ Contains additional definitions for environment variables; see
 above.
 .Pp
 .It Pa ~/.ssh/id_ecdsa
-.It Pa ~/.ssh/id_ecdsa_sk
 .It Pa ~/.ssh/id_ed25519
-.It Pa ~/.ssh/id_ed25519_sk
 .It Pa ~/.ssh/id_rsa
 Contains the private key for authentication.
 These files
@@ -1614,9 +1602,7 @@ generating the key which will be used to encrypt the
 sensitive part of this file using AES-128.
 .Pp
 .It Pa ~/.ssh/id_ecdsa.pub
-.It Pa ~/.ssh/id_ecdsa_sk.pub
 .It Pa ~/.ssh/id_ed25519.pub
-.It Pa ~/.ssh/id_ed25519_sk.pub
 .It Pa ~/.ssh/id_rsa.pub
 Contains the public key for authentication.
 These files are not
--- old/ssh_config.5
+++ new/ssh_config.5
@@ -474,8 +474,6 @@ The default is:
 .Bd -literal -offset indent
 ssh-ed25519,ecdsa-sha2-nistp256,
 ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,
-sk-ssh-ed25519@openssh.com,
-sk-ecdsa-sha2-nistp256@openssh.com,
 rsa-sha2-512,rsa-sha2-256
 .Ed
 .Pp
@@ -1130,16 +1128,10 @@ ssh-ed25519-cert-v01@openssh.com,
 ecdsa-sha2-nistp256-cert-v01@openssh.com,
 ecdsa-sha2-nistp384-cert-v01@openssh.com,
 ecdsa-sha2-nistp521-cert-v01@openssh.com,
-sk-ssh-ed25519-cert-v01@openssh.com,
-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
 rsa-sha2-512-cert-v01@openssh.com,
 rsa-sha2-256-cert-v01@openssh.com,
 ssh-ed25519,
 ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,
-sk-ssh-ed25519@openssh.com,
-sk-ecdsa-sha2-nistp256@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256@openssh.com,
 rsa-sha2-512,rsa-sha2-256
 .Ed
 .Pp
@@ -1178,16 +1170,10 @@ ssh-ed25519-cert-v01@openssh.com,
 ecdsa-sha2-nistp256-cert-v01@openssh.com,
 ecdsa-sha2-nistp384-cert-v01@openssh.com,
 ecdsa-sha2-nistp521-cert-v01@openssh.com,
-sk-ssh-ed25519-cert-v01@openssh.com,
-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
 rsa-sha2-512-cert-v01@openssh.com,
 rsa-sha2-256-cert-v01@openssh.com,
 ssh-ed25519,
 ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,
-sk-ecdsa-sha2-nistp256@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256@openssh.com
-sk-ssh-ed25519@openssh.com,
 rsa-sha2-512,rsa-sha2-256
 .Ed
 .Pp
@@ -1791,16 +1777,10 @@ ssh-ed25519-cert-v01@openssh.com,
 ecdsa-sha2-nistp256-cert-v01@openssh.com,
 ecdsa-sha2-nistp384-cert-v01@openssh.com,
 ecdsa-sha2-nistp521-cert-v01@openssh.com,
-sk-ssh-ed25519-cert-v01@openssh.com,
-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,
 rsa-sha2-512-cert-v01@openssh.com,
 rsa-sha2-256-cert-v01@openssh.com,
 ssh-ed25519,
 ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,
-sk-ssh-ed25519@openssh.com,
-sk-ecdsa-sha2-nistp256@openssh.com,
-webauthn-sk-ecdsa-sha2-nistp256@openssh.com,
 rsa-sha2-512,rsa-sha2-256
 .Ed
 .Pp
